Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -167,7 +167,7 @@ void ImageEditor::paint_event(GUI::PaintEvent& event)
     }
 
     if (m_active_layer && m_show_active_layer_boundary)
-        painter.draw_rect(content_to_frame_rect(m_active_layer->relative_rect()).to_rounded<int>().inflated(2, 2), Color::Black);
+        painter.draw_rect(content_to_frame_rect(m_active_layer->relative_rect()).to_type<int>().inflated(2, 2), Color::Black);
```

Comment:
This arguably gives you a worse result since instead of rounding, you're now flooring the value. Isn't the issue that there is a cast to int somewhere else?

Reply:
Everywhere else in `ImageEditor` uses truncation when converting float coordinates to int. Either via `to_type<int>()` or implicitly in methods like `frame_to_content_position()`, `frame_to_content_rect()` and others. The idea of this change was to make layer boundary drawing consistent with everything else.

Reply:
@GMTA - I've just pushed an alternative implementation, that fixes `Image::paint_into()` directly. I believe this better shows the issues that led to the problem.

I have also squashed together the commits that fix layer painting and layer boundary painting because, as you say, they each give a worse result when applied individually, but are both necessary for a correct result.